galsim/sed.py

```diff
@@ -818,6 +818,49 @@ def calculateSeeingMomentRatio(self, bandpass, alpha=-0.2, base_wavelength=500):
             return galsim.integ.int1d(lambda w: weight(w) * kernel(w),
                                       bandpass.blue_limit, bandpass.red_limit) / flux
 
+    def sampleWavelength(self, nphotons, bandpass, seed=0, npoints=256):
+        """ Sample 'nphoton' random wavelengths from the SED (and bandpass).
+
+        @param nphotons  Number of samples (photons) to randomly draw.
+        @param bandpass  Bandpass through which object is being imaged.
+        @param seed      Something that can seed a BaseDeviate: a long int seed or another
```

Comment:
This is my fault, since I think I told you to call this parameter `seed`, but to make this more consistent with other functions/classes that need random deviates, we typically call this parameter `rng`, and nominally it should just be an existing BaseDeviate object.  None is still allowed, in which case a default BaseDeviate object is constructed automatically for the user.